### PR TITLE
[MFMA] Parameterize MFMA layout with non k dimension size

### PR DESCRIPTION
**Pull request by binarman**
This PR adds more attributes to MFMA layout to align with dot-rocMLIR branch.

https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/4660

### CODE REVIEW

**Comment by zhanglx13** [comment]
I just browsed the PR and here is my comment

I think our plan is
- you keep the current layout of MfmaEncodingAttr
- later the dot-rocMLIR branch will change its layout to what you have

Originally, rocMLIR and CK are using different block layout like
MLIR_vs_CK_block_layout
dot-rocMLIR branch needs the `xdlopsPerWave` since each wave computes a number of tiles that are adjacent to each other. However, in CK's layout, tiles handled by each wave are interleaved. And I believe this is also the case in the triton-mlir branch.

**Comment by binarman** [comment]
@zhanglx13 
Thank you for diagrams!
I understand the difference between CK and dtriton-mlir layouts.

So, let me check if I understand you correct:

Soon dot-rocMLIR will remove `xdlopsPerWave` field, so I should not add it.

P.s.

> in CK's layout, tiles handled by each wave are interleaved

Could you elaborate this?

**Comment by binarman** [comment]
If you want to add more parameters in future dot-rocMLIR, please tell.
I would like to add them in this PR, if possible

**Comment by zhanglx13** [comment]
No, my plan is to remove `xdlopsPerWave`

File: python/test/unit/language/test_core_amd.py [hunk 1]
```diff
@@ -2143,11 +2146,13 @@ def __init__(self, size_per_thread, threads_per_warp, warps_per_cta, order):
     def __str__(self):
         return f"#triton_gpu.blocked<{{sizePerThread={self.sz_per_thread}, threadsPerWarp={self.threads_per_warp}, warpsPerCTA={self.warps_per_cta}, order={self.order}}}>"
 
+
```
Comment (binarman) [inline python/test/unit/language/test_core_amd.py:2149]: some of these changes are added by auto formatting tools

File: lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp [hunk 2]
```diff
@@ -152,9 +152,16 @@ class BlockedToMFMA : public mlir::RewritePattern {
     auto ctx = oldAType.getContext();
 
     triton::gpu::MfmaEncodingAttr mfmaEnc;
+
+    int64_t nonKDim = 32;
+
     auto warpsPerTile = warpsPerTileMI200(dotOp, retShape, numWarps);
+
+    auto mPerWave = retShape[0] / warpsPerTile[0];
+    auto nPerWave = retShape[1] / warpsPerTile[1];
```
Comment (zhanglx13) [inline lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp:161]: These two are not used, are they?

**Review by zhanglx13 — APPROVED** [review]
I think this is a valid first step toward 16x16 mfma instruction support. Approved.
Please remove `mPerWave` and `nPerWave` since they are not used.

File: lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp (hunk 2 above)
Comment (binarman) [reply lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp]: fixed, thank you!

**Pull request merged**